jdbc/src/java/org/apache/hive/jdbc/HiveStatement.java

```diff
@@ -197,6 +198,12 @@ private void closeStatementIfNeeded() throws SQLException {
     } catch (SQLException e) {
       throw e;
     } catch (Exception e) {
+      if (e instanceof TApplicationException && ((TApplicationException)e)
```

Comment:
Please do not use `instanceof` here, just create a new `catch` block for `TApplicationException` types.